### DISCUSSION TITLE
How to render track and background as separate videos?

### DISCUSSION THREAD

**Discussion by pedjas** [gpx-animator/gpx-animator#598] [Q&A] [answered]
Is it possible to render track (with solid background color) and map background as separate videos?

That would allow using them in video editing software and add more graphical elements and effects.

**Comment by pentool** [comment, marked as answer]
You can't render two videos at the same time. You have to render out the track first on a solid background, then you have to render out the map background. You can also render the track as white on a black background and use that as alpha.

**Comment by pedjas** [comment]
Well, I did not mean to render them at the same time, just to have an option to set to create two videos, one after another.
 
I can create track video with custom background but have problems creating background map video without track. There is quite a lot settings alterations to hide everything and, yet, I did not find way to make track totally invisible.

Alto there is an issue of having project saved. I have to change a lot of project settings to get two different videos, but I cannot have those settings saved within project so when I need to repeat export I have to manually do all configuration alterations.

I meant to have option like "Export track and background as separate videos", or "Export track only" and "Export background only" so GPX Animator can automatically do what is necessary to have track invisible in one case and background invisible in other case and create two videos.

I guess that would be valuable to everyone who needs to have them as two separate layers in video editing so additional effects and processing can be applied to track and background map separately.

**Reply by pentool** [reply]
If you don't want the track to be visible, just set the "Line Width" and "Pre-Draw Line Width" (on the Track tab) to zero (0). In that case only the little round marker will be visible. If you don't want the marker, set its size (Marker Size) to like 0.1 or 0.01 and you won't see it.

Also, I'm not sure what is the issue with saving the parameters. You select File > Save and you can save all values. You save one set for the track only and another set for the map/background only project. Then all you have to do is open whichever you want.

**Comment by Phatman113** [comment]
I'm jumping in because I just found this tool and it looks awesome!  I've been digging around, but I can't seem to figure out how to do part of what pedjas is asking about.  How do I render a track with a transparent background?  I want to use this as an overlay on a separate video, but I can't figure out how to export with anything but a solid background.  Is that possible with this tool?  

Thanks to everyone who's making this tool possible, it seems really perfect!

**Reply by pedjas** [reply]
You can set background to green and then use chroma key masking in video editing software.
